modin/pandas/test/dataframe/test_join_sort.py

```diff
@@ -329,6 +329,81 @@ def test_merge(test_data, test_data2):
         modin_df.merge("Non-valid type")
 
 
+def test_merge_with_mi_columns():
+    modin_df1, pandas_df1 = create_test_dfs(
+        {
+            ("col0", "a"): [1, 2, 3, 4],
+            ("col0", "b"): [2, 3, 4, 5],
+            ("col1", "a"): [3, 4, 5, 6],
+        }
+    )
+
+    modin_df2, pandas_df2 = create_test_dfs(
+        {
+            ("col0", "a"): [1, 2, 3, 4],
+            ("col0", "c"): [2, 3, 4, 5],
+            ("col1", "a"): [3, 4, 5, 6],
+        }
+    )
+
+    res = modin_df1.merge(modin_df2, on=[("col0", "a")])
+    ref = pandas_df1.merge(pandas_df2, on=[("col0", "a")])
+    df_equals(res, ref)
```

Comment:
```suggestion
    eval_general(
        modin_df1,
        pandas_df1,
        lambda df: df.merge(modin_df2, on=[("col0", "a")])
        if isinstance(df, pd.DataFrame)
        else df.merge(pandas_df2, on=[("col0", "a")])
    )
```
everywhere

Reply:
I didn't use `eval_general` on purpose here, IMO it looks clumsy when it tests functions that require these if-else statements.  We're not supposed to test any exception-matching here, so using `eval_general` is not beneficial here.

Reply:
okay, found a way to make `eval_general` calls prettier:
```python
        eval_general(
            (modin_df1, modin_df2),
            (pandas_df1, pandas_df2),
            lambda dfs: dfs[0].merge(dfs[1], on=on),
        )
```
so now using it everywhere in my tests